ExampleProtocols.thy

```diff
@@ -10,46 +10,50 @@ begin
 
 (* Definition 4.2 *)
-fun later :: "(message * state) \<Rightarrow> state"
+fun later :: "params \<Rightarrow> (message * state) \<Rightarrow> state"
   where
-    "later (m, \<sigma>) = {m'. m' \<in> \<sigma> \<and>  m \<in> justification m'}"
+    "later params (m, \<sigma>) = {m' \<in> M params. m' \<in> \<sigma> \<and>  m \<in> justification m'}"
 
 (* Definition 4.3: Messages From a Sender *)
-fun from_sender :: "(validator * state) \<Rightarrow> state"
+fun from_sender :: "params \<Rightarrow> (validator * state) \<Rightarrow> state"
   where
-    "from_sender (v, \<sigma>) = {m. m \<in> \<sigma> \<and> sender m = v}"
+    "from_sender params (v, \<sigma>) = {m \<in> M params. m \<in> \<sigma> \<and> sender m = v}"
 
 (* Definition 4.4: Message From a Group *)
-fun from_group :: "(validator set * state) \<Rightarrow> state"
+fun from_group :: "params \<Rightarrow> (validator set * state) \<Rightarrow> state"
   where
-    "from_group (v_set, \<sigma>) = {m. m \<in> \<sigma> \<and> sender m \<in> v_set}"
+    "from_group params (v_set, \<sigma>) = {m \<in> M params. m \<in> \<sigma> \<and> sender m \<in> v_set}"
 
 (* Definition 4.5 *)
-fun later_from :: "(message * validator * state) \<Rightarrow> state"
+fun later_from :: "params \<Rightarrow> (message * validator * state) \<Rightarrow> state"
   where
-    "later_from (m, v, \<sigma>) = later (m, \<sigma>) \<inter> from_sender (v, \<sigma>)"
+    "later_from params (m, v, \<sigma>) = later params (m, \<sigma>) \<inter> from_sender params (v, \<sigma>)"
 
 (* Definition 4.6: Latest Message *)
-definition latest_message :: "state \<Rightarrow> (validator \<Rightarrow> state)"
+definition latest_message :: "params \<Rightarrow> state \<Rightarrow> (validator \<Rightarrow> state)"
   where
-    "latest_message \<sigma> v = {m. m \<in> from_sender (v, \<sigma>) \<and> later_from (m, v, \<sigma>) = \<emptyset>}"
+    "latest_message params \<sigma> v = {m \<in> M params. m \<in> from_sender params (v, \<sigma>) \<and> later_from params (m, v, \<sigma>) = \<emptyset>}"
 
 (* Definition 4.7: Latest message driven estimator *)
 (* TODO *)
 
 (* Definition 4.8: Latest Estimates *)
-definition latest_estimates :: "state \<Rightarrow> validator \<Rightarrow> consensus_value set"
+definition latest_estimates :: "params \<Rightarrow> state \<Rightarrow> validator \<Rightarrow> consensus_value set"
   where
-    "latest_estimates \<sigma> v = {est m | m. m \<in> latest_message \<sigma> v}"
+    "latest_estimates params \<sigma> v = {est m | m. m \<in> M params \<and> m \<in> latest_message params \<sigma> v}"
 
 (* Definition 4.9: Latest estimate driven estimator *)
 (* TODO *)
 
 (* Lemma 5: Non-equivocating validators have at most one latest message *)
 lemma non_equivocating_validators_have_at_most_one_latest_message:
-  "\<forall> params v. v \<in> V params \<and> v \<notin> equivocating_validators \<sigma> \<Longrightarrow> card (latest_message \<sigma> v) \<le> 1"
```

Comment:
@yudetamago 
I think it should not `Longrightarrow` here?

Reply:
You are right